src/storage-client/src/controller.rs

```diff
@@ -1939,6 +1948,40 @@ where
         }
     }
 
+    async fn snapshot_stats(
+        &self,
+        id: GlobalId,
+        as_of: Antichain<Self::Timestamp>,
+    ) -> Result<SnapshotStats<Self::Timestamp>, StorageError> {
+        // WIP DNM registering a new handle and throwing it away each time this
```

Comment:
Realized I’m probably overthinking things here. I think we should be able to just toss a clone of Machine down inside the SinceHandle impl

Reply:
got it working!